benches/reissue.rs

```diff
@@ -146,9 +146,9 @@ fn bench_reissue_100_to_1(c: &mut Criterion) {
 fn generate_dbc_of_value(
     amount: Amount,
     rng: &mut (impl RngCore + CryptoRng),
-) -> Result<(SpentBookNodeMock, Dbc)> {
+) -> Result<(mock::SpentBookNode, Dbc)> {
```

Comment:
not sure about the general plan of this `mock` and the real production code.
However, normally what we do is here we use a `normal` struct, say `SpentBookNode`
Then within the implementation of `SpentBookNode`, we use the `mock` feature flag to switch beteen the real production code and mock code.
Though currently we only have mock code ?

same comment to other similar usage.

Reply:
Here the meaning of "mock" is a little bit different.

Mock objects represent system components that are intended to be implemented outside of sn_dbc.

So for example, with SpentBook, there is a logical concept of a SpentBook Node, but sn_dbc only implements SpentProof in its public API.  So basically sn_dbc doesn't care what a SpentBook Node looks like, what it's API is, or anything.  It only cares that somehow there exists something which can produce a SpentProof.  So this is a very loose coupling... there is not even a Trait for a Spentbook.

Still, there is a need to write test cases that simulate a fully functional system.  So the mock SpentBookNode (and other mock objects) exist for this purpose, and for prototyping basic logic.   But they would never be used in production code.

Reply:
but there is still `mock` stuff appears in the production code? 
like in the `src/dbc.rs` file ?

If it does like what you say only for the testing, then better put it as a `test_utility` ?
like what we have for `safe_network`, where there is `utility` under the `tests` folder

Reply:
@maqi Looks like all the `mock` usages in `src/dbc.rs` etc. are just in test cases. (we have unit tests inline in the source files).

The mock's are also used in the examples and benches as well, e.g. the mint-repl and reissue benchmarks makes heavy use of the mocks